onnxscript/ir/_convenience/__init__.py

```diff
@@ -373,5 +373,6 @@ def replace_nodes_and_values(
             graph_or_function.outputs[idx] = replacement_mapping[graph_or_function_output]
 
     # insert new nodes after the index node
-    graph_or_function.insert_after(insertion_point, new_nodes)
+    if new_nodes:
```

Comment:
This is allowing empty new_nodes. I can create another function if you think it's out of the scope in this function. @justinchuby

Reply:
I want to reason about this function a bit more carefully because the renaming behavior currently is also no ideal. Can you copy the logic to the pass and use it privately for now? I intend to freeze this function before the onnx ir release.

Reply:
That makes sense. We just want to replace uses of Y... by X..., and remove the node computing Y...

Reply:
Created another function without adding new node.